plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/s3/S3FileSystemExchangeStorage.java

```diff
@@ -455,6 +459,22 @@ private static AwsCredentialsProvider createAwsCredentialsProvider(ExchangeS3Con
             return StaticCredentialsProvider.create(AwsBasicCredentials.create(config.getS3AwsAccessKey(), config.getS3AwsSecretKey()));
         }
 
+        if (config.getS3IamRole().isPresent()) {
```

Comment:
Can you validate config here that different configuration options are mutually exclusive?
* both `config.getS3AwsAccessKey()` and `config.getS3AwsSecretKey()` must be either set or unset
* if accessKey and secretKey are set then other auth related config options are not set.

Reply:
I'm not sure if we need to do that because `TrinoS3FileSystem` didn't perform such checks. Also they are not really mutually exclusive, it's just that priority is different (static credentials > iamRole > default credentials)

Reply:
I think the fact that TS3FS does not do that is an omission. 
Also unless I am missing sth they are mutually exclusive. If you set S3 Access and Secret key it does not matter what you set in IamRole as you would not get to line in code when you use that.
You would return in  `return StaticCredentialsProvider.create(AwsBasicCredentials.create(config.getS3AwsAccessKey(), config.getS3AwsSecretKey()));` right?

Hence if user provides both access/secretKey and IAM role it is fishy, I would throw in such case.

Reply:
OK, let me add the validations

Reply:
Thanks. I made the assertions a bit more explicit and used IllegalArgumentException instead of `verify` (verify should rather be used for consistency assertions - for things that would never happen if system is working correctly).
PTAL if I did not mess anything up.

Reply:
I'm fine with this, but I think my version is more concise. We can merge it still

Reply:
Yeah it is :) But I think actionable error messages are more important than concise code. 
I am also dumb - and reading concise code is hard for me :P